### ISSUE TITLE
Style/CollectionMethods does not detect `inject` -> `reduce` in certain cases

### ISSUE THREAD

**Issue by dgollahon** [rubocop-hq/rubocop#9262]
## Expected behavior

```ruby
[1, 2].inject(:+)     # =>  Style/CollectionMethods: Prefer reduce over inject.
[1, 2].inject(0, :+)  # =>  Style/CollectionMethods: Prefer reduce over inject.
[1, 2].inject(0, &:+) # =>  Style/CollectionMethods: Prefer reduce over inject.
```

## Actual behavior

```ruby
[1, 2].inject(:+)     # => Nothing is flagged
[1, 2].inject(0, :+)  # => Nothing is flagged
[1, 2].inject(0, &:+) # => Nothing is flagged
```
Note that the following works correctly:

```ruby
[1, 2].inject(&:+) 				   # => Style/CollectionMethods: Prefer reduce over inject.
[1, 2].inject(0) { |a, b| a + b }  # => Style/CollectionMethods: Prefer reduce over inject.
```

## Steps to reproduce the problem

Add an `.inject` with  to some code and run `rubocop`.

## RuboCop version

rubocop -V

```
warning: parser/current is loading parser/ruby27, which recognizes
warning: 2.7.2-compliant syntax, but you are running 2.7.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.7.1 x86_64-darwin19)
```
